docs/modules/rag.mdx

```diff
@@ -0,0 +1,181 @@
+</Tip>
+
+## Philosophy
+
+BeeAI Framework's approach to RAG emphasizes **integration over invention**. Rather than building RAG components from scratch, we provide seamless adapters for proven, production-ready solutions from leading platforms like LangChain and Llama-Index.
+
+This philosophy offers several advantages:
+- **Leverage existing expertise** - Use battle-tested implementations
+- **Faster time-to-market** - Focus on your application logic, not infrastructure
+- **Community support** - Benefit from extensive documentation and community
+- **Flexibility** - Switch between providers as needs evolve
+
+## RAG Components
+
+The following table outlines the key RAG components available in the BeeAI Framework:
+
+| Component | Description | Compatibility | Future Compatibility |
+|-----------|-------------|---------------|---------------------|
+| **Document Loaders** | Responsible for loading content from different formats and sources such as PDFs, web pages, and structured text files | WIP | BeeAI, LangChain |
+| **Text Splitters** | Splits long documents into workable chunks using various strategies, e.g. fixed length or preserving context | WIP | BeeAI, LangChain |
+| **Document** | The basic data structure to house text content, metadata, and relevant scores for retrieval operations | BeeAI | - |
+| **Vector Store** | Used to store document embeddings and retrieve them based on semantic similarity using embedding distance | LangChain | BeeAI, Llama-Index |
+| **Document Processors** | Used to process and refine documents during the retrieval-generation lifecycle including reranking and filtering | Llama-Index | - |
+
+## Dynamic Module Loading
+
+BeeAI Framework provides a dynamic module loading system that allows you to instantiate RAG components using string identifiers. This approach enables configuration-driven architectures and easy provider switching.
+
+The `from_name` method uses the format `provider:ClassName` where:
+- `provider` identifies the integration module (e.g., "beeai", "langchain")
+- `ClassName` specifies the exact class to instantiate
+
+<Tip>
+Dynamic loading enables you to switch between different vector store implementations without changing your application code - just update the configuration string.
+</Tip>
+
+### BeeAI Vector Store
+
+```python
+from beeai_framework.backend.vector_store import VectorStore
+from beeai_framework.adapters.watsonx.backend.embedding import WatsonxEmbeddingModel
+
+# Initialize embedding model
+embedding_model = WatsonxEmbeddingModel(
+    model_id="ibm/slate-125m-english-rtrvr-v2",
+    project_id=os.getenv("WATSONX_PROJECT_ID"),
+    api_key=os.getenv("WATSONX_APIKEY"),
+    base_url=os.getenv("WATSONX_URL"),
+)
```

Comment:
Use `EmbeddingModel.from_name(...)` for easier use (api_key and base_url will be loaded from env automatically).